### ISSUE TITLE
[$250] Android - Disabled bank detail input field flickers when navigating back

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#51397]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.53-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Issue was found when executing this PR:** https://github.com/Expensify/App/pull/51234
**Email or phone of affected tester (no customers):** nathanmulugetatesting+1937@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Open NewDot hybrid app
2. Create a workspace
3. Enable workflows
4. Start manual bank account connection flow
5. On the first step enter routing number and account number and click on next
6. Go back to the previous page and observe the input fields

## Expected Result:
Disabled input fields without flickering is observed

## Actual Result:
Disabled input fields flicker

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/92d47cc8-a8e6-4c23-95cb-3a897ca1034f

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021871546683215081160</li>
        <li>Upwork Job ID: 1871546683215081160</li>
        <li>Last Price Increase: 2025-01-14</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ishpaul777</details>

**Comment by youssef-lr** [comment]
Checking

**Comment by youssef-lr** [comment]
No update, low priority

**Comment by jliexpensify** [comment]
@youssef-lr - want to make this a `Weekly` if it's lower priority?

**Comment by jliexpensify** [comment]
Making this a weekly so we're not pinged

**Comment by jliexpensify** [comment]
Not overdue

**Comment by jliexpensify** [comment]
@youssef-lr I think this is a #quality project?

**Comment by youssef-lr** [comment]
Seems like a regular bug to me @jliexpensify

**Comment by jliexpensify** [comment]
Which project would you associate it with @youssef-lr?

**Comment by youssef-lr** [comment]
I think #expense fits.

**Comment by jliexpensify** [comment]
Not overdue, waiting on Youssef

**Comment by youssef-lr** [comment]
Let's make this external

**Comment by rohit9625** [comment]
Is the Android Hybrid App different from what external contributors can build from the source? I can reproduce this issue on the Standalone Android App.

**Comment by rohit9625** [comment]
Hey @youssef-lr, I've solved the flickerring issue when navigating back to step 1:-

https://github.com/user-attachments/assets/b079d3bf-dfa0-46d8-a6c6-456661d58082

But there's a minor shift in the text that you can see in the video. Similar issue was addressed here #26753.
However, I've figured out the cause of that also. Look at the code below:-
![image](https://github.com/user-attachments/assets/a0cd499f-426b-4d01-a493-550496cbd2a3)

I beleive that extra height was meant for IOS only but it is also affecting Android devices. We can use native android method to pass that value as 0.
Please correct me if I'm wrong.

**Comment by rohit9625** [comment]
Great @daledah, it solved the issue. I didn't know about that parameter, thanks. However, my solution was a little bit complex I guess.

```bash
function Manual({onNext}: ManualProps) {
    const [reimbursementAccount, result] = useOnyx(ONYXKEYS.REIMBURSEMENT_ACCOUNT);
        const [hasBankAccountData, setHasBankAccountData] = useState(!!(reimbursementAccount?.achData?.bankAccountID ?? ''));

    useEffect(() => {
        if (result.status === 'loaded' && reimbursementAccount) {
            setHasBankAccountData(!!(reimbursementAccount?.achData?.bankAccountID ?? ''));
        }
    }, [result, reimbursementAccount]);

**Comment by jliexpensify** [comment]
Bump @abdulrahuman5196 for some reviews!

**Comment by abdulrahuman5196** [comment]
Hi, will check in a day. Out for holiday.

**Comment by abdulrahuman5196** [comment]
Hi @jliexpensify , Sorry, I am still not able to get this through. I don't want to delay this issue. Kindly re-apply the label for different C+. Unassigning myself.

**Comment by jliexpensify** [comment]
Hi @ishpaul777 - can you help us with this issue? We've got some proposals - thanks!

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Disabled bank detail input field flickers when navigating back

### What is the root cause of that problem?

When we return from the personal info section, the value of `reimbursementAccount` changes three times. The first and last times, `reimbursementAccount` has a value, while the second time, `reimbursementAccount` is `undefined`.

https://github.com/Expensify/App/blob/ebba9671a2ad2684ad19e7c9f9b43350c0cdd056/src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx#L25

With that, the condition to disable the input is determined by the `reimbursementAccount` data, where it alternates between disable being `true`, then `false`, and then `true` again, which has caused this issue to occur.

https://github.com/Expensify/App/blob/ebba9671a2ad2684ad19e7c9f9b43350c0cdd056/src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx#L36

https://github.com/Expensify/App/blob/ebba9671a2ad2684ad19e7c9f9b43350c0cdd056/src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx#L89

https://github.com/Expensify/App/blob/ebba9671a2ad2684ad19e7c9f9b43350c0cdd056/src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx#L102

The `reimbursementAccount` is `undefined`, which indicates that the response data is incomplete.

### What changes do you think we should make in order to solve the problem?

To resolve this issue, we need to wait for the data to fully load by checking the isLoadingOnyx status for reimbursementAccount. If isLoadingOnyx is true, we should delay rendering the component or handle it appropriately.

https://github.com/Expensify/App/blob/ebba9671a2ad2684ad19e7c9f9b43350c0cdd056/src/pages/ReimbursementAccount/BankInfo/substeps/Manual.tsx#L66

```diff
    const [reimbursementAccount, reimbursementAccountResult] = useOnyx(ONYXKEYS.REIMBURSEMENT_ACCOUNT);
    ...
    if (isLoadingOnyxValue(reimbursementAccountResult)) {
        return null; // Render a loading spinner or return null while the data is loading
    }
```

By implementing this check, we ensure that the component does not attempt to access or process incomplete data, thereby preventing any potential issues caused by partially loaded or inconsistent states.


### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by jliexpensify** [comment]
Bumping @ishpaul777 for reviews please! Thanks.

**Comment by jliexpensify** [comment]
Hmm the PR was from October - lets get Applause to re-test

**Comment by jliexpensify** [comment]
Waiting on a re-test, labels updated

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by jliexpensify** [comment]
Nice, that confirms what @ishpaul777 is seeing - closing this.
